pymatgen/analysis/piezo_sensitivity.py

```diff
@@ -0,0 +1,602 @@
+    ops = sga(struc).get_symmetry_operations(cartesian = True)
+    structure = phonopy.get_phonopy_structure(struc)
+
+    pnstruc = pn.Phonopy(structure, np.eye(3), np.eye(3))
+    supercell = pnstruc.get_supercell()
+    primitive = pnstruc.get_primitive()
+
+    
+    converter = dyntofc.DynmatToForceConstants(primitive, supercell)
+    pnstruc.set_force_constants(fcm)
+    dyn = pnstruc.get_dynamical_matrix_at_q([0,0,0])
+    dyn = np.reshape(dyn, (len(struc),3,len(struc),3)).swapaxes(1,2)
+    passed = []
+    relations = []
+    for i in range(len(dyn)):
+        for j in range(i, len(dyn)):
+            eig1, vecs1 = np.linalg.eig(dyn[i][j])
+    #         eig2, vecs2 = np.linalg.eig(dyn[j][i])
+#             print(i,j, eig)
+            index = np.argsort(eig1)
+            neweig = np.real([eig1[index[0]],eig1[index[1]],eig1[index[2]]])
+            passed.append([i,j,np.real(neweig)])
+    #         passed.append([j,i,eig2])
+            for k in range(len(passed)):
+                if np.allclose(neweig,passed[k][2], atol = 1e-05):
+                    relations.append([i,j, passed[k][0], passed[k][1]])
+                    break
+    operations = []
+    for i in range(len(relations)):
+        good = 0
+        operations.append(relations[i])
+        if relations[i][2] == relations[i][0] and relations[i][1] == relations[i][3]:
+            operations[i].append([])
+            transpose = []
+            
+            for j in range(len(ops)):
+                new = ops[j].transform_tensor(dyn[relations[i][2]][relations[i][3]])
+                if np.allclose(new, dyn[relations[i][1]][relations[i][0]], atol = 1e-05):
+                    transpose.append(ops[j])
+            operations[i].append(transpose)        
+            transpose = []
+            continue
+        
+        for j in range(len(ops)):
+            new = ops[j].transform_tensor(dyn[relations[i][2]][relations[i][3]])
+            if np.allclose(new, dyn[relations[i][0]][relations[i][1]], atol = 1e-05):
+                operations[i].append(ops[j])
+                good = 1
+
+            elif np.allclose(new, dyn[relations[i][0]][relations[i][1]].T, atol = 1e-05):
+                operations[i].append(ops[j])
+                temp1 = operations[i][2]
+                temp2 = operations[i][3]
+                operations[i][2] = temp2
+                operations[i][3] = temp1
+                good = 1
+            if good ==1:
+                transpose = []
+            
+                for j in range(len(ops)):
+                    new = ops[j].transform_tensor(dyn[relations[i][2]][relations[i][3]])
+                    if np.allclose(new, dyn[relations[i][1]][relations[i][0]], atol = 1e-05):
+                        transpose.append(ops[j])
+                operations[i].append(transpose)
+                break
+        
+    return operations
+
+                    
+def rand_piezo(struc, fcm, anumiter = 10):
```

Comment:
This should probably a submethod of piezo tensor, but for now, we can leave this here.